.circleci/config.yml

```diff
@@ -210,6 +218,60 @@ jobs:
                   [ -n "<< pipeline.parameters.PACKAGE_INSTALL_COMMAND >>" ] && eval "<< pipeline.parameters.PACKAGE_INSTALL_COMMAND >>"
                   eval "<< pipeline.parameters.BUILD_COMMAND >>"
 
+      - when:
+          condition:
+            equal: [true, "<< pipeline.parameters.IS_NEXT_JS >>"]
+          steps:
+            - run:
+                name: Build Next.js Project
+                working_directory: ~/project/repository
+                command: |
+                  set -e
+                  [ -f ~/project/.env ] && source ~/project/.env
+                  [ -f ~/project/.env.ef_circleci ] && source ~/project/.env.ef_circleci
+
+                  # Detect package manager
+                  if [ -f "yarn.lock" ]; then
```

Comment:
a lot of projects have started using pnpm but there's no pnpm file. balancer also uses pnpm https://github.com/balancer/frontend-monorepo/tree/main/apps/frontend-v3